engine/source/public/core/asset_manager.hpp

```diff
@@ -0,0 +1,75 @@
+﻿#pragma once
+#include "core/asset.hpp"
+
+#include <memory>
+#include <unordered_map>
+
+namespace lumina
+{
+    class Asset;
+    class Engine;
+
+    class AssetManager
+    {
+    private:
+        friend class Engine;
+        std::unordered_map<uint32_t, std::shared_ptr<Asset>> assets;
+
+    public:
+        template <typename T, typename... Args>
+        std::shared_ptr<T> CreateAsset(Args&&... args)
+        {
+            //Check if T is derived from Asset
```

Comment:
I configured clang to always have a single space after the comment slashes, but maybe that didn't properly work...
so instead of 
```c++
//comment attached to slashes
```
I had configured
```c++
// comment has a single space after slashes
```